packages/cosmic-swingset/src/chain-main.js

```diff
@@ -450,14 +464,22 @@ export default async function main(progname, args, { env, homedir, agcc }) {
       portNums.lien = action.lienPort;
     }
 
-    // Ensure that initialization has completed.
-    blockingSend = await (blockingSend || launchAndInitializeSwingSet(action));
-
-    if (action.type === AG_COSMOS_INIT) {
-      // console.error('got AG_COSMOS_INIT', action);
-      return true;
-    }
-
-    return blockingSend(action);
+    swingsetActive = true;
+
+    return Promise.resolve(
+      // Ensure that initialization has completed.
+      blockingSend || launchAndInitializeSwingSet(action),
+    )
+      .then(doBlockingSend => {
+        if (action.type === AG_COSMOS_INIT) {
+          // console.error('got AG_COSMOS_INIT', action);
+          return true;
+        } else {
+          return doBlockingSend(action);
+        }
```

Comment:
My style preference, but YMMV:
```suggestion
        }
        return doBlockingSend(action);
```